### PR TITLE
Make observability tests part of standard test suite

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**
In order to run tests regarding observability in the standard test scenario include them in `npm test`

Changes proposed in this pull request:

- Move Grafana tests to designated test file _3-observability.js_
- Move Prometheus tests to designated test file _3-observability.js_
- Include _3-observability.js_ tests in `npm test`
- Refactor helpers
- Some renaming on getting-started-guide

**Related issue(s)**
* https://github.com/kyma-project/kyma/issues/12692

### CODE REVIEW

**Review by shorim — COMMENTED** [review]
The code from `monitoring-test/monitoring-test.js` has been moved to `monitoring/prometheus.js`. However, you forgot now to delete the `monitoring-test/monitoring-test.js` file. I think now we can actually delete the entire directory `monitoring-test`, right?

**Review by shorim — COMMENTED** [review]
grafana-related helpers have been moved from `helpers.js` to `grafana.js`. However, we still have prometheus-related helpers in `helpers.js`. Why is this distinction?

I would suggest one of the 2 options:
1- To have `grafana.js`, `grafana-helpers.js`, `prometheus.js` and`prometheus-helpers.js` where `grafana.js` and `prometheus.js` only have the main assertions.
2- Move the remaining helper functions in `helpers.js` to `prometheus.js`.

I would prefer option 1 to have better separation.

**Review by shorim — COMMENTED** [review]
Since the `3-observability.js` contains only tests for prometheus and grafana, I would suggest to rename it to `3-monitoring.js`. This is just to avoid the misunderstanding that it contains all observability-related tests, as the logging test is actually integrated within `1-commerce-mock.js`

**Comment by lindnerby** [comment]
> grafana-related helpers have been moved from `helpers.js` to `grafana.js`. However, we still have prometheus-related helpers in `helpers.js`. Why is this distinction?
> 
> I would suggest one of the 2 options: 1- To have `grafana.js`, `grafana-helpers.js`, `prometheus.js` and`prometheus-helpers.js` where `grafana.js` and `prometheus.js` only have the main assertions. 2- Move the remaining prometheus-related helper functions in `helpers.js` to `prometheus.js` and delete `helpers.js`.
> 
> I would prefer option 1 to have better separation.

Opted to remove helpers for now. Once we merge all observability tests in one folder let's think about how to extract.

**Review by shorim — COMMENTED** [review]
We should also remove `monitoring/readme.md` and this one liner https://github.com/kyma-project/kyma/blob/main/tests/fast-integration/package.json#L24, since we removed the `monitoring-test` directory.

**Review by NHingerl — DISMISSED** [review]
👍 renamed folder

File: tests/fast-integration/monitoring/grafana.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    await waitForPodWithLabel("app", "grafana", namespace);
+}
+
+async function assertGrafanaRedirectsExist() {
+    if (getEnvOrDefault("KYMA_MAJOR_VERSION", "2") === "2") {
+        await checkGrafanaRedirectsInKyma2();
+    } else {
+        await checkGrafanaRedirectsInKyma1();
+    }
+}
```
Comment (skhalash) [inline tests/fast-integration/monitoring/grafana.js:33]: I am wondering if we have ever run those tests for Kyma 1. @rakesh-garimella, do you know. If it's not relevant. let us get rid of it.

Comment (lindnerby) [reply tests/fast-integration/monitoring/grafana.js:33]: Would like to do that then in the separate clean-up PR I'm preparing. This one is mostly about restructuring and including tests in the default npm test

Comment (rakesh-garimella) [reply tests/fast-integration/monitoring/grafana.js:33]: yeah we should handle it in a separate PR as we should also cleanup from various pipelines where it is set.

**Review by rakesh-garimella — APPROVED** [review]

File: tests/fast-integration/monitoring/prometheus.js [hunk 2]
```diff
@@ -140,14 +140,13 @@ function shouldIgnoreTarget(target) {
...
     var alertNamesToIgnore = [
-        // Watchdog is an alert meant to ensure that the entire alerting pipeline is functional, so it should always be firing,
+        // Watchdog is an alert meant to ensure that the entire alerting pipeline is functional
         "Watchdog",
         // Scrape limits can be exceeded on long-running clusters and can be ignored
         "ScrapeLimitForTargetExceeded",
-
-        // Overcommitting means that a cluster with at least 1 node taken down doesn't have enough resources to run all the pods
-        // It's fine in an e2e test scenario since the clusters are usually deliberately created small to save money
+        // If a cluster with at least 1 node taken down doesn't have enough resources to run all the pods which is fine for e2e test scenarios
```
Comment (NHingerl) [inline tests/fast-integration/monitoring/prometheus.js:149]:
```suggestion
        // If a cluster with at least 1 node taken down doesn't have enough resources to run all the pods, which is fine for e2e test scenarios
```

Not sure if I'm parsing this sentence correctly. It's pretty long, but still seems to miss the bit that says what happens if a cluster (...) doesn't have enough resources.

Comment (lindnerby) [reply tests/fast-integration/monitoring/prometheus.js:149]: It's just a comment for us to know why we ignore the alert :) wanted to shorten it a bit

Comment (NHingerl) [reply tests/fast-integration/monitoring/prometheus.js:149]: Aaah of course, it follows to line 143 "alerts we don't care about".
```suggestion
        // If a cluster with at least 1 node taken down doesn't have enough resources to run all the pods; that's fine for e2e test scenarios
```

Comment (lindnerby) [reply tests/fast-integration/monitoring/prometheus.js]: Changed it a bit to be short and concise

**Review by shorim — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**